### PR TITLE
Bryan Hoke - UI Design Patterns Challenge

### PR DESCRIPTION
**Pull request by BryanHoke**

### CODE REVIEW

File: MVP Module/Source/Feed UI/Controllers/FeedViewController.swift [hunk 1]
```diff
@@ -33,6 +35,14 @@ public final class FeedViewController: UITableViewController, UITableViewDataSou
 		}
 	}
 
+	func display(_ viewModel: FeedErrorViewModel) {
+		guard let message = viewModel.message else {
+			errorView?.hideMessage()
+			return
+		}
+		errorView?.show(message: message)
```
Comment (thiagopenna) [inline MVP Module/Source/Feed UI/Controllers/FeedViewController.swift:43]: `if/else` is more idiomatic and easier to read in this context.

For example: "If there's an error message, show it. Else, hide any existing message."

Instead of: "guard against a `nil` error message. Else, hide the current message and return. Show the error message."
```suggestion
		if let message = viewModel.message {
			errorView?.show(message: message)
		} else {
			errorView?.hideMessage()
		}
```

File: MVVM Module/Source/Feed Presentation/FeedViewModel.swift [hunk 2]
```diff
@@ -20,12 +20,17 @@ final class FeedViewModel {
 
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onFeedLoadError: Observer<String?>?
```
Comment (thiagopenna) [inline MVVM Module/Source/Feed Presentation/FeedViewModel.swift:23]: Suggestion: rename `onFeedLoadError` to `onErrorStateChange`, since this property is also used when there are no errors.

File: MVVM Module/Source/Feed UI/Controllers/FeedViewController.swift [hunk 3]
```diff
@@ -32,6 +34,13 @@ public final class FeedViewController: UITableViewController, UITableViewDataSou
 				self?.refreshControl?.endRefreshing()
 			}
 		}
+		viewModel?.onFeedLoadError = { [weak self] message in
+			guard let message = message else {
+				self?.errorView?.hideMessage()
+				return
+			}
+			self?.errorView?.show(message: message)
```
Comment (thiagopenna) [inline MVVM Module/Source/Feed UI/Controllers/FeedViewController.swift:42]: `if/else` is more idiomatic and easier to read in this context.

For example: "If there's an error message, show it. Else, hide any existing message."

Instead of: "guard against a `nil` error message. Else, hide the current message and return. Show the error message."
```suggestion
			if let message = viewModel.message {
				self?.errorView?.show(message: message)
			} else {
				self?.errorView?.hideMessage()
			}
```

**Review by thiagopenna — COMMENTED** [review]
Hi @BryanHoke,

This is Penna, I am an instructor at Essential Developer.

Congratulations, you've passed the challenge and got your third belt stripe ⚪️③!

✅ All tests are passing
✅ Your code is clean and handles edge cases
✅ The UI and animations look and behave as expected
✅ Light and Dark mode support
✅ Added an extra language

**[Update your status](https://iosleadessentials.slack.com/archives/CTAR0LVMM/p1629389971188200) in the community now with the following emoji to celebrate your achievement: `:white-belt-3:`**

Keep up the great work! There's one last challenge ahead. Remember, we're here to help you progress and get the most out of this program.

See you in the community and in the final challenge 🙂

---

**P.S. I left some final review comments. Feel free to push changes.**
